.github/workflows/test.yml

```diff
@@ -1,6 +1,12 @@
 name: Run tests
 run-name: ${{ github.actor }} is running tests
-on: push
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - main
```

Comment:
I need to do a little more research to make sure I understand the security implications of running CI against PRs

Reply:
The github actions security model seems a little convoluted and I'm not sure I fully understand it, but I went into the settings and set it to "always require approval" to run on PRs, which I think should keep things pretty safe.

Reply:
Requires approval is always safe, the threat model is secure by default. Things that break the threat model are basically `pull_request_target` or anything that allows use of secrets from PRs or write access to repo contents via PRs. [The whole PyPi release flow can be done secretless these days via OIDC](https://docs.pypi.org/trusted-publishers/using-a-publisher/#github-actions).

https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/